apps/src/code-studio/components/SoundListEntry.jsx

```diff
@@ -81,7 +89,9 @@ class SoundListEntry extends React.Component {
       this.props.soundsRegistry.unmuteURLs();
       this.props.soundsRegistry.playURL(this.props.soundMetadata.sourceUrl, {
         onEnded: () => {
-          this.setState({isPlaying: false});
+          if (this._isMounted) {
+            this.setState({isPlaying: false});
+          }
```

Comment:
Rather than the anti-pattern that you mentioned, you should be able to manage this using redux.